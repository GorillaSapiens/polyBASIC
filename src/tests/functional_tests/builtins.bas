a = 100+6/2
print "100+6/2=",a
a = 100/5/2
print "100/5/2=",a
a = abs(-5)
print "abs(-5)=",a
a = atn(0)
print "atn(0)=",a
a = chr(65)
print "chr(65)=",a
a = cos(0)
print "cos(0)=",a
a = dbl(#4'2/3)
print "dbl(#4'2/3)=",a
a = exp(1.0)
print "exp(1.0)=",a
a = int(#-4'2/3)
print "int(#-4'2/3)=",a
a = rat(1.333333333)
print "rat(1.333333333)=",a
a = "|" & str(22) & "|"
print "str(22)=",a
a = log(exp(1))
print "log(exp(1))=",a
a = rnd(4.5)
print "rnd(4.5)=",a
REM a = randomize
REM print "randomize=",a
REM a = rnd(4.5)
REM print "rnd(4.5)=",a
a = sgn(-12)
print "sgn(-12)=",a
a = sgn(#0)
print "sgn(#0)=",a
a = sgn(15.2)
print "sgn(15.2)=",a
a = sin(3.14159265358/2)
print "sin(3.14159265358/2)=",a
a = sqr(#2)
print "sqr(#2)=",a
print "tab(20)",tab(20),"hello!"
