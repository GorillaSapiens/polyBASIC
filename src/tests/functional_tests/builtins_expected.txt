100+6/2= 103
100/5/2= 10
abs(-5)= 5
atn(0)= 0.0
chr(65)= A
chr(129421)= ü¶ç
cos(0)= 1.0
dbl(#4'2/3)= 4.66667
exp(1.0)= 2.71828
int(#-4'2/3)= -5
rat(1.333333333)= #1'1/3
rat(4.4623655914)=#4'43/93= #4'43/93
rat(2.0625)=#2'1/16= #1'1/16
str(22)= |22|
log(exp(1))= 1.0
rnd(4.5)= 3.78084
sgn(-12)= -1
sgn(#0)= 0
sgn(15.2)= 1
sin(3.14159265358/2)= 1.0
sqr(#2)= 1.41421
tab(20) [20G hello!
